### ISSUE TITLE
Zero variable arguments should be allowed in C23

### ISSUE THREAD

**Issue by Halalaluyafail3** [llvm/llvm-project#84495]
Clang doesn't allow zero variable arguments in macro invocations of macros with named arguments and then variable arguments. Here is an example program (https://godbolt.org/z/xbacsGqbM):
```c
#include<stdarg.h>
void foo(...){
    va_list v;
    va_start(v);
    va_arg(v,int(*)[5]);
    va_end(v);
}
int main(void){
    foo(&(int[]){1,2,3,4,5});
}
```
Clang complains here about not having a comma in `va_start` and wants `va_start(v,)`. This restriction was removed in C23 so this code should be allowed without any complaint.

**Comment by Sirraide** [comment]
This is only with `-pedantic-errors`. Looks like another case of an extension that is now legal in C23 and which we shouldn’t warn about anymore.
